presto-spi/src/main/java/io/prestosql/spi/connector/Name.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class Name
+{
+    private final String name;
+    private final boolean isDelimited;
+
+    public static Name createNonDelimitedName(String name)
+    {
+        return new Name(name, false);
+    }
+
+    @JsonCreator
+    public Name(@JsonProperty("name") String name, @JsonProperty("isDelimited") boolean isDelimited)
+    {
+        this.name = requireNonNull(name, "Name is null");
+        this.isDelimited = isDelimited;
+    }
+
+    @JsonProperty("name")
+    public String getName()
+    {
+        return name;
+    }
+
+    @JsonProperty("isDelimited")
+    public boolean isDelimited()
+    {
+        return isDelimited;
+    }
+
+    public String getLegacyName()
```

Comment:
We shouldn't have any calls to this method in the engine. This is to ease transition to the new API for connectors. Also, we should mark it as `@Deprecated`